addons/base_address_extended/tests/test_street_fields.py

```diff
@@ -15,44 +16,76 @@ def setUpClass(cls):
         cls.env.ref('base.ch').write({'street_format': 'header %(street_name)s, %(street_number)s - %(street_number2)s trailer'})
         cls.env.ref('base.mx').write({'street_format': '%(street_name)s %(street_number)s/%(street_number2)s'})
 
-    def create_and_assert(self, partner_name, country_id, street, street_name, street_number, street_number2):
-        partner = self.Partner.create({'name': partner_name + '-1', 'street': street, 'country_id': country_id})
-        self.assertEqual(partner.street_name or '', street_name, 'wrong street name for %s: %s' % (partner_name, partner.street_name))
-        self.assertEqual(partner.street_number or '', street_number, 'wrong house number for %s: %s' % (partner_name, partner.street_number))
-        self.assertEqual(partner.street_number2 or '', street_number2, 'wrong door number for %s: %s' % (partner_name, partner.street_number2))
-        partner = self.Partner.create({
-            'name': partner_name + '-2',
-            'street_name': street_name,
-            'street_number': street_number,
-            'street_number2': street_number2,
-            'country_id': country_id,
+    def assertStreetVals(self, partner, street_data):
+        for key, val in street_data.items():
+            if isinstance(partner[key], models.BaseModel):
+                self.assertEqual(partner[key].id, val, 'Wrongly formatted street field %s: expected %s, received %s' % (key, val, partner[key]))
+            else:
+                self.assertEqual(partner[key], val, 'Wrongly formatted street field %s: expected %s, received %s' % (key, val, partner[key]))
+
+    def test_partner_create(self):
```

Comment:
`""" Will test the compute and inverse methods of street fields when creating partner records. """`